src/lock_commit/node.rs

```diff
@@ -49,15 +51,72 @@ pub enum State {
 
 use State::*;
 
+#[derive(Clone)]
+pub struct NodeReceiverHandler {
+    /// Used to forward incoming TCP messages to the node
+    pub network_sender: Sender<(Command, oneshot::Sender<Result<Option<Vec<u8>>>>)>,
+}
+
 #[async_trait]
-impl MessageHandler for Node {
+impl MessageHandler for NodeReceiverHandler {
+    /// When a TCP message is received, interpret it as a node::Message and forward it to the node task.
+    /// Send the node's response back through the TCP connection.
     async fn dispatch(&mut self, writer: &mut Writer, bytes: Bytes) -> Result<()> {
         let request: Command = bincode::deserialize(&bytes)?;
-        info!("{}: Received request {:?}", self.socket_address, &request);
+        log::info!("Received request {:?}", request);
+
+        let (reply_sender, reply_receiver) = oneshot::channel();
+        self.network_sender.send((request, reply_sender)).await?;
+        let reply = reply_receiver.await?.map_err(|e| e.to_string());
+
+        let reply = bincode::serialize(&reply)?;
+        log::info!("Sending response {:?}", reply);
+
+        let _ = writer.send(reply.into()).await?;
+
+        Ok(())
+    }
+}
 
+impl Node {
+    pub fn new(
+        peers: Vec<SocketAddr>,
+        db_path: &str,
+        address: SocketAddr,
+        timer_start: Arc<RwLock<Instant>>,
+    ) -> Self {
+        Self {
+            store: Store::new(db_path).unwrap(),
+            peers: peers,
+            sender: SimpleSender::new(),
+            current_view: Arc::new(RwLock::new(0)),
```

Comment:
Now that the node is not cloned when we receive TCP messages, we can remove the Arcs and the RwLocks/Mutexes, since there will only be one struct object for the duration of the program. We can merge this PR and leave that as a future work item (please create a task for that), or you can try removing it here. Notice that you will have to change some of the code in the logic so that it does not get locks, but it will make the code nicer.